hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/OperationSubsystemStart.java

```diff
@@ -31,28 +33,43 @@
 
     @Override
     public void execute(OperationContext opContext, ModelNode model) throws OperationFailedException {
+
+        final AtomicReference<Thread> newThread = new AtomicReference<>();
+
         try {
             ServiceName name = SubsystemExtension.SERVICE_NAME;
             ServiceRegistry serviceRegistry = opContext.getServiceRegistry(true);
             MonitorService service = (MonitorService) serviceRegistry.getRequiredService(name).getValue();
 
             boolean restart = model.get("restart").asBoolean(false);
-            if (restart) {
-                LOGGER.debug("Asked to restart the Hawkular Monitor service. Will stop it, then restart it now.");
-                service.stopMonitorService();
-            } else {
-                LOGGER.debug("Asked to start the Hawkular Monitor service");
-            }
-            if (!service.isMonitorServiceStarted()) {
-                Thread newThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
+            long delay = model.get("delay").asLong(0L);
+
+            newThread.set(new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    if (delay > 0) {
+                        try {
+                            Thread.sleep(delay);
+                        } catch (Exception e) {
+                            newThread.get().interrupt();

```

Comment:
see my comment below - this should simply be "return;" here in the catch clause. If the sleep is interruptted, you don't want it to interrupt itself and keep going! You want it to return and which effectively kills the thread.